src/common/tusb_fifo.h

```diff
@@ -52,18 +52,74 @@ extern "C" {
 #define tu_fifo_mutex_t  osal_mutex_t
 #endif
 
+/* Write/Read index is always in the range of:
```

Comment:
@PanRe I added implementation note for fifo. If you have time, please review this to see if I miss anything.